### PR TITLE
Validate and Release Creation

### PR DESCRIPTION
**Pull request by diabawada**
Description:

1. Versions File:
In this pull request, we focus on managing our software versions effectively. We'll first decide the version number to be generated.

2. Validate Version Pipeline:
validate_version pipeline plays a crucial role in ensuring that the configured version in the file will not conflict with an existing version. It helps us maintain version consistency and prevent conflicts down the line.

3. Merge to Main Pipeline:
After the validation is successful, we'll merge the changes into the main branch. This will trigger our merge_main pipeline, which takes care of generating a new tag and creating a release for our software. This ensures that our releases are always in sync with the main branch.

### CODE REVIEW

File: .github/workflows/validate_version.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Validate Release
+
+on:
+  deployment:
```
Comment (niv-ez) [inline .github/workflows/validate_version.yaml]: why on deployment? who triggers it?
not better using workflow dispatch?

File: version [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Use Semantic versioning only. Please update the version number before opening a pull request.
+v1.0.0
```
Comment (niv-ez) [inline version]: start with v0.0.1

File: .github/scripts/common.sh
Comment (omriezra) [inline .github/scripts/common.sh:1]: please review akeyless-environments/.github/scripts/git/git_tag_validation.sh

File: .github/scripts/validate_release_version.sh
Comment (omriezra) [inline .github/scripts/validate_release_version.sh:1]: Please review akeyless-environments/.github/scripts/git/git_tagging.sh

File: .github/workflows/merge_main.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        id: create_release
+        uses: softprops/action-gh-release@v1
+        with:
+          tag_name: $version
+          name: Release-${{ env.version }}
+          draft: false
+          prerelease: false
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
```
Comment (omriezra) [inline .github/workflows/merge_main.yaml:41]: you alerady have acton for this .github/workflows/release_creation.yaml

File: .github/workflows/merge_main.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+        id: create_release
+        uses: softprops/action-gh-release@v1
+        with:
+          tag_name: $version
+          name: Release-${{ env.version }}
+          draft: false
+          prerelease: false
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (omriezra) [inline .github/workflows/merge_main.yaml:40]: all of this not related to the release, please review our git tag and release flow

File: .github/workflows/validate_version.yaml (hunk 1 above)
Comment (liorhasson) [reply .github/workflows/validate_version.yaml]: @diabawada only `pull_request` required here

File: .github/workflows/release_creation.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Create Release
+        id: create_release
+        uses: softprops/action-gh-release@v1
+        with:
+          tag_name: ${{ github.ref_name }}
+          name: Release-${{ github.ref_name }}
+          draft: false
+          prerelease: false
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (omriezra) [inline .github/workflows/release_creation.yaml]: this creates empty content, is that ok?

File: .github/workflows/validate_version.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Fetch history
+        run: git fetch --prune --unshallow
+
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Validate Release Version
+        run: .github/scripts/validate_release_version.sh
```
Comment (omriezra) [inline .github/workflows/validate_version.yaml:23]: this performs as your previous action, i don't under stand this duplications, please explain

File: version
Comment (omriezra) [inline version:1]: we are not holding version in file, remove and check our current flows.

**Review by omriezra — DISMISSED** [review]

File: .github/workflows/validate_version.yaml (hunk 1 above)
Comment (diabawada) [reply .github/workflows/validate_version.yaml]: fixed

File: version (hunk 2 above)
Comment (diabawada) [reply version]: fixed

File: .github/workflows/merge_main.yaml (hunk 3 above)
Comment (diabawada) [reply .github/workflows/merge_main.yaml:41]: yes I merged this file to the release creation pipeline

File: .github/workflows/validate_version.yaml (hunk 6 above)
Comment (diabawada) [reply .github/workflows/validate_version.yaml:23]: This runs only if the user opens a PR to check if and  inform the user to update the version number

File: .github/workflows/release_creation.yaml (hunk 5 above)
Comment (diabawada) [reply .github/workflows/release_creation.yaml]: Do you mean generate an empty release?

File: version
Comment (diabawada) [reply version:1]: This was Lior request!

File: .github/workflows/merge_main.yaml (hunk 4 above)
Comment (diabawada) [reply .github/workflows/merge_main.yaml:40]: Can you explain, please?

File: .github/scripts/common.sh
Comment (liorhasson) [reply .github/scripts/common.sh:1]: @omriezra this is not blocker for now

File: .github/scripts/validate_release_version.sh
Comment (liorhasson) [reply .github/scripts/validate_release_version.sh:1]: @omriezra this is not blocker for now

File: version
Comment (liorhasson) [reply version:1]: @omriezra this is fine. we use the same logic in [terraform provider](https://github.com/akeyless-community/terraform-provider-akeyless)

this is a simple way for community contribution

also in `helm-chart` repo we manually modifying the version in `chart.yaml` file

**Comment by liorhasson** [comment]
@diabawada please add `CONTRIBUTING.md` file, take this as an example: https://github.com/actions/checkout/blob/main/CONTRIBUTING.md

**Review by omriezra — COMMENTED** [review]
Clean all your tests tags and releases created

**Review by liorhasson — APPROVED** [review]

**Pull request merged**